### PR TITLE
Plfm 5829

### PR DESCRIPTION
**Pull request by brucehoff**

### CODE REVIEW

File: lib/models/src/main/java/org/sagebionetworks/repo/model/GroupMembersDAO.java [hunk 1]
```diff
@@ -14,6 +14,14 @@
 	public List<UserGroup> getMembers(String principalId) 
 			throws DatastoreException, NotFoundException;
 	
+	/**
+	 * Return the subset of groups in which the given principal is a member
+	 * @param principalId
+	 * @param groupIds
+	 * @return
+	 */
+	public List<String> queryGroups(String principalId, List<String> groupIds);
```
Comment (brucehoff) [inline lib/models/src/main/java/org/sagebionetworks/repo/model/GroupMembersDAO.java]: filterUserGroups

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImpl.java [hunk 2]
```diff
@@ -122,6 +128,14 @@
 		UserGroupUtils.copyDboToDto(dbos, members);
 		return members;
 	}
+	
+	@Override
+	public List<String> queryGroups(String principalId, List<String> groupIds) {
+		MapSqlParameterSource param = new MapSqlParameterSource();
```
Comment (brucehoff) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImpl.java:136]: validate principalId is not null
if groupIds is null or empty then return empty list

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImplTest.java [hunk 3]
```diff
@@ -85,6 +86,14 @@ public void testGetters() throws Exception {
...
 		List<UserGroup> groups = groupMembersDAO.getUsersGroups(testUserOne.getId());
 		assertEquals("No groups initially", 0, groups.size());
+		
+		List<String> groupList = new ArrayList<String>();
+		groupList.add(testGroup.getId());
+		List<String> groupIds = groupMembersDAO.queryGroups(testUserOne.getId(), groupList);
+		assertTrue(groupIds.isEmpty());
+		groupList.add("99999");
+		groupIds = groupMembersDAO.queryGroups(testUserOne.getId(), groupList);
+		assertTrue(groupIds.isEmpty());
```
Comment (brucehoff) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImplTest.java:123]: try passing nulls and try passing empty group list

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/claimprovider/TeamClaimProvider.java [hunk 4]
```diff
@@ -47,22 +47,8 @@ public Object getClaim(String userId, OIDCClaimsRequestDetails details) {
...
-		ListWrapper<TeamMember> teamMembers = teamDAO.listMembers(numericTeamIds, Collections.singletonList(Long.parseLong(userId)));
-		Set<String> result = new HashSet<String>();
-		if (teamMembers.getList()!=null) {
-			for (TeamMember teamMember : teamMembers.getList()) {
-				result.add(teamMember.getTeamId());
-			}
-		}
+		List<String> actualTeamIds = groupMembersDAO.queryGroups(userId, new ArrayList<String>(requestedTeamIds));
+		Set<String> result = new HashSet<String>(actualTeamIds);
 		return result;
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/claimprovider/TeamClaimProvider.java]: eliminate the extraneous function call

File: services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImplUnitTest.java [hunk 5]
```diff
@@ -112,7 +113,7 @@
 	private UserProfileManager userProfileManager;
 	
 	@Mock
-	private TeamDAO mockTeamDAO;
+	private GroupMembersDAO groupMembersDAO;
```
Comment (brucehoff) [inline services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImplUnitTest.java]: mockGroupMembersDAO

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImplTest.java [hunk 6]
```diff
@@ -86,14 +87,53 @@ public void testGetters() throws Exception {
...
+		// an empty group list is filtered to an empty list
+		groupIds = groupMembersDAO.filterUserGroups(testUserOne.getId(), Collections.EMPTY_LIST);
+		assertTrue(groupIds.isEmpty());
+		
+		// if the user isn't in the group, an empty list is returned
+		groupIds = groupMembersDAO.filterUserGroups(testUserOne.getId(), groupList);
+		assertTrue(groupIds.isEmpty());
+
+		// if the user isn't in the group, an empty list is returned
+		groupIds = groupMembersDAO.filterUserGroups(testUserOne.getId(), groupList);
```
Comment (jay-hodgson) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImplTest.java:118]: delete (line 117 and 118)?

**Pull request merged**